shell/platform/linux/fl_key_event_plugin.h

```diff
@@ -27,23 +27,35 @@ G_DECLARE_FINAL_TYPE(FlKeyEventPlugin,
 /**
  * fl_key_event_plugin_new:
  * @messenger: an #FlBinaryMessenger.
+ * @response_callback: the callback to call when a response is received.  If not
+ *                     given (nullptr), then the default response callback is
+ *                     used.
+ * @channel_name: the name of the channel to send key events on. If not given
+ *                (nullptr), then the standard key event channel name is used.
+ *                Typically used for tests to send on a test channel.
  *
  * Creates a new plugin that implements SystemChannels.keyEvent from the
  * Flutter services library.
  *
  * Returns: a new #FlKeyEventPlugin.
  */
-FlKeyEventPlugin* fl_key_event_plugin_new(FlBinaryMessenger* messenger);
+FlKeyEventPlugin* fl_key_event_plugin_new(
+    FlBinaryMessenger* messenger,
+    GAsyncReadyCallback response_callback = nullptr,
+    const char* channel_name = nullptr);
```

Comment:
Wouldn't it make more sense for the mock engine to allow setting up an arbitrary channel to echo, rather than passing in an arbitrary channel name here, so that the changes for testing as, as much as possible, in the test code rather than the real code?

Reply:
OK @stuartmorgan , I've added an engine constructor that will allow overriding of the channel to send requests on, and used that when setting up the mock engine.  Is that what you were thinking of? I still have to pass the response callback, because that's specific to the key event plugin, and it needs to terminate the event loop.

Reply:
I think this is worse than what you had, sorry. Per discussion yesterday, I forgot there's no actual mock engine at this point, only a mock renderer.

Let's go back to what you had before, and longer term we'll find a better way of doing this kind of testing. (Probably using the embedder.h layer interposing that I want to write for testing all of the desktop embeddings.)

Reply:
OK, I reverted it back to what I had before.